### ISSUE TITLE
Bug: Move off of abandoned `sanity-codegen`

### ISSUE THREAD

**Issue by jpmcb** [open-sauced/landing-page#328]
The Sanity Codegen project has been archived: https://github.com/ricokahler/sanity-codegen

> As the sole maintainer of sanity-codegen, I found myself unable to dedicate the time and attention it truly deserved. That's why I'm thrilled to announce the arrival of Sanity TypeGen! This project is a complete replacement for sanity-codegen and is maintained in-house by the same team that maintains GROQ itselt

It'd also be good to move off of this since there are existing Dependabot security alerts related to transient dependencies carried in `sanity-codegen` (likely since it hasn't been updated in some time).

---

I took a brief look at using the new Sanity TypeGen and it won't be trivial unfortunately: we're still on an old version of Sanity studio:

https://github.com/open-sauced/landing-page/blob/6768fb928b97c89622787085b7d94a12d02bad66/sanity/package.json#L17

which was deprecated this last December:

- https://github.com/open-sauced/landing-page/issues/185

---

TLDR:

- [ ] Upgrade to Sanity v3
- [ ] Remove `sanity-codegen`
- [ ] Utilize Sanity TypeGen
